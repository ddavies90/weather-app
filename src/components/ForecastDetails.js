import React from "react";
import PropTypes from "prop-types";

const ForecastDetails = ({ forecast }) => {
  const { date, humidity, temperature, wind } = forecast;

  const convertedDate = new Date(date).toLocaleDateString("default", {
    weekday: "short",
    day: "numeric",
    month: "short"
  });

  return (
    <div className="forecast-details">
      <div className="forecast-details__date">Date: {convertedDate}</div>
      <div className="forecast-details__maxtemp">
        Max: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__mintemp">
        Min: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__windspeed">
        Wind Speed: {wind.speed}mph
      </div>
      <div className="forecast-details__winddir">
        Wind Dir: {wind.direction}
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    id: PropTypes.number,
    date: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string
    }),
    humidity: PropTypes.number,
    description: PropTypes.string,
    icon: PropTypes.string
  }).isRequired
};

export default ForecastDetails;
