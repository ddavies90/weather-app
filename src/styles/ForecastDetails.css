.forecast-details {
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  margin-top: 3rem;
  font-size: 1.5rem;
}

.forecast-details__date {
  font-size: 2rem;
  background-color: yellow;
  padding: 0.5rem 0.7rem;
  color: red;
}

.forecast-details__maxtemp {
  padding-top: 0.3rem;
}

.forecast-details__key {
  color: cyan;
}
