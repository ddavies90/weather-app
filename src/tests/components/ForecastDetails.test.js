import React from "react";
import { render, screen } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("<ForecastDetails />", () => {
  const validProps = {
    date: 1525132800000,
    temperature: {
      max: 12,
      min: 3
    },
    humidity: 58,
    wind: {
      speed: 10,
      direction: "se"
    }
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct props with appropriate formatting", () => {
    render(<ForecastDetails forecast={validProps} />);

    expect(screen.getByText("Tue, 1 May")).toBeInTheDocument();
    expect(screen.getByText("12°C")).toBeInTheDocument();
    expect(screen.getByText("3°C")).toBeInTheDocument();
    expect(screen.getByText("58%")).toBeInTheDocument();
    expect(screen.getByText("10mph")).toBeInTheDocument();
    expect(screen.getByText("SE")).toBeInTheDocument();
  });
});
