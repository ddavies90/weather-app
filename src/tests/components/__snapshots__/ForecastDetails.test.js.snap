// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastDetails /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Tue, 1 May
    </div>
    <div
      class="forecast-details__maxtemp"
    >
      <span
        class="forecast-details__key"
      >
        Max: 
      </span>
      12°C
    </div>
    <div
      class="forecast-details__mintemp"
    >
      <span
        class="forecast-details__key"
      >
        Min: 
      </span>
      3°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      <span
        class="forecast-details__key"
      >
        Humidity: 
      </span>
      58%
    </div>
    <div
      class="forecast-details__windspeed"
    >
      <span
        class="forecast-details__key"
      >
        Wind Speed: 
      </span>
      10mph
    </div>
    <div
      class="forecast-details__winddir"
    >
      <span
        class="forecast-details__key"
      >
         Direction: 
      </span>
      SE
    </div>
  </div>
</DocumentFragment>
`;
